doxygen/mbedtls.doxyfile

```diff
@@ -710,7 +710,11 @@ EXCLUDE_SYMLINKS       = NO
 # against the file with absolute path, so to exclude all test directories
 # for example use the pattern */test/*
 
-EXCLUDE_PATTERNS       = *_internal.h *_wrap.h
+EXCLUDE_PATTERNS       = cipher_internal.h \
+                         md_internal.h \
+                         pk_internal.h \
+                         ssl_internal.h \
+                         *_wrap.h
```

Comment:
Which `_internal.h` files are we additionally allowing now? Is it clear to folks adding new `_internal.h` files that they won't get excluded automatically anymore?

Reply:
I think there has been a shift in what `_internal.h` is used for: it used to mean "don't include this in your application, it's only for internal library use", but in recently added occurrences it means "don't include this in your application, it's only for implementers of alternative crypto implementations". I wish we had distinct names for those two meanings but it's two late for that. Anyway, it looks like new `_internal.h` files will most likely fall in that second category, so should indeed be included by default. (Ideally we could have different doxyfiles for end-users and alt-implementers.)

Reply:
I've added a comment to explain this.